src/build/functions/server.ts

```diff
@@ -87,7 +87,7 @@ const writeHandlerManifest = async (ctx: PluginContext) => {
 
 const writePackageMetadata = async (ctx: PluginContext) => {
   await writeFile(
-    join(ctx.serverHandlerRootDir, 'package.json'),
+    join(ctx.serverHandlerRootDir, '.netlify', 'package.json'),
     JSON.stringify({ type: 'module' }),
   )
```

Comment:
I'm not quite sure if this is a correct fix (or rather if this is full fix) - we definitely shouldn't be setting `type: 'module'` for entire handler directory in every case (which is what cause the problem right now). 

We definitely should either set it for our runtime modules we copy to handler (which is what this PR does right now) OR make runtime emit `.mjs` files and not `.js` ones to `dist`.

But part that I'm not sure about is wether `type: 'module'` should be set in root of handler in case user use this mode? I'll run some experiments here, but in case someone is familiar - I'm happy for any heads up on that.

Reply:
So I did update it since this comment - Next.js does output their own `package.json` in root of standalone which we were ignoring when copying from it. With my changes we copy it over (this is really just copy of site's package.json so it inherit site's `type` essentially) + create package.json just for runtime modules (possibly this could be avoided if we emitted `.mjs` files for runtime modules, but my initial attempt at that resulted in problems so I didn't pursue this too much)